logs_sdk/lib/opentelemetry/sdk/logs/logger_provider.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Logs
+      # The SDK implementation of OpenTelemetry::Logs::LoggerProvider.
+      class LoggerProvider < OpenTelemetry::Logs::LoggerProvider
+        attr_reader :resource, :log_record_processors
+
+        EMPTY_NAME_ERROR = 'LoggerProvider#logger called without '\
+          'providing a logger name.'
+        FORCE_FLUSH_ERROR = 'unexpected error in ' \
+          'OpenTelemetry::SDK::Logs::LoggerProvider#force_flush'
+
+        # Returns a new LoggerProvider instance.
+        #
+        # @param [optional Resource] resource The resource to associate with
+        #   new LogRecords created by {Logger}s created by this LoggerProvider.
+        # @param [optional Array] log_record_processors The
+        #   {LogRecordProcessor}s to associate with this LoggerProvider.
+        #
+        # @return [OpenTelemetry::SDK::Logs::LoggerProvider]
+        def initialize(
+          resource: OpenTelemetry::SDK::Resources::Resource.create,
+          log_record_processors: []
+        )
+          @log_record_processors = log_record_processors
+          @mutex = Mutex.new
+          @resource = resource
+          @stopped = false
+        end
+
+        # Creates an {OpenTelemetry::SDK::Logs::Logger} instance.
+        #
+        # @param [optional String] name Instrumentation package name
+        # @param [optional String] version Instrumentation package version
+        #
+        # @return [OpenTelemetry::SDK::Logs::Logger]
+        def logger(name = nil, version = nil)
+          name ||= ''
+          version ||= ''
+
+          OpenTelemetry.logger.warn(EMPTY_NAME_ERROR) if name.empty?
```

Comment:
> In the case where an invalid name (null or empty string) is specified, a working Logger MUST be returned as a fallback rather than returning null or throwing an exception, its name SHOULD keep the original invalid value, and a message reporting that the specified value is invalid SHOULD be logged.
[source](https://opentelemetry.io/docs/specs/otel/logs/sdk/#loggerprovider)

This logic is drawn from [TracerProvider#tracer](https://github.com/open-telemetry/opentelemetry-ruby/blob/4347e0f47bb0d6a4870f1e9311f20d3d344fb34a/sdk/lib/opentelemetry/sdk/trace/tracer_provider.rb#L51-L56). The null or empty string will not stop InstrumentationScope from being generated during Logger initialization.